lib.c

```diff
@@ -159,13 +159,52 @@ static int is_filtered(struct track_info *ti)
 	return 0;
 }
 
+static bool track_exists(struct track_info *ti)
+{
+	struct rb_node *node;
+	struct artist *artist;
+	struct album *album;
+	struct tree_track *track;
+
+	if (!ti->title)
+		return false;
+
+	const char *ti_artist_name = tree_artist_name(ti);
+	rb_for_each_entry(artist, node, &lib_artist_root, tree_node) {
+		if (strcmp(artist->name, ti_artist_name) == 0)
+			break;
+	}
+	if (!artist)
+		return false;
+
+	const char *ti_album_name = tree_album_name(ti);
+	rb_for_each_entry(album, node, &artist->album_root, tree_node) {
+		if (strcmp(album->name, ti_album_name) == 0)
```

Comment:
For consistency, this should probably use `special_album_cmp` instead. This function will also check the date and will use the sort name instead if present.

Reply:
Thanks for the pointer. It seems to directly use those functions I'd have to construct temporary album / artist objects - doable, although perhaps just slightly ugly / inefficient.

- The overall album date is defined by the newest track, so adding it here is tricky. I could add it below as a `ti` check, but I'm anyway not convinced it's needed. The most likely cause of a different date (but equal everything else) is that the date is blank or filled in wrong, rather than it not being a duplicate.
- I wouldn't care if the sort name is _different_, but I suppose theoretically some tracks could set a sort name and leave "name" blank so it is worth accounting for.
- Looking at this function, we could very easily be using the "collkey_" versions of everything to be case insensitive, and I probably should.

Reply:
> some tracks could set a sort name and leave "name" blank

I'm gonna disagree with yesterday me. Nobody should be doing this.